app/logical/post_query_builder.rb

```diff
@@ -118,6 +118,12 @@ def tags_match(tags, relation)
 
   def metatags_match(metatags, relation)
     metatags.each do |metatag|
+      if metatag.negated
+        metatag.name = "fav" if metatag.name == "ordfav"
+        metatag.name = "favgroup" if metatag.name == "ordfavgroup"
+        metatag.name = "pool" if metatag.name == "ordpool"
+      end
+
       clause = metatag_matches(metatag.name, metatag.value, quoted: metatag.quoted)
       clause = clause.negate_relation if metatag.negated
```

Comment:
Right now, `negate_relation` does `unscoped.where(all.where_clause.invert.ast)` (From [searchable.rb](https://github.com/danbooru/danbooru/blob/2fff56302705bc0e5ee723c72a851c51dbf150d4/app/logical/concerns/searchable.rb#L16))
I don't know the compatibility with other queries or how taxing it would be to change it but `unscoped.where.not(id: all.pluck(:id))` sounds good in my head.
Is this something to consider?

Reply:
Funny thing is that it works without explicitly `pluck`ing the IDs.

Reply:
See https://discord.com/channels/310432830138089472/310846683376517121/934147187175194674 and the messages after it. Summary: this creates a chunky query that cucks all other order statements so replacing the metatags is preferable.